src/components/Button/PrimaryButton.tsx

```diff
@@ -0,0 +1,151 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+
+import { InjectedProps, withTheme } from '../../hocs/withTheme'
+
+import { hoverable } from '../../hocs/hoverable'
+import { isTouchDevice } from '../../libs/ua'
+
+type Tag = 'button' | 'a'
+type Size = 's'
+
+interface ClickEvent {
+  preventDefault: () => void
+}
+
+interface BaseProps {
+  size?: Size
+  children?: React.ReactNode
+  className?: string
+  prefix?: React.ReactNode
+  suffix?: React.ReactNode
+  square?: boolean
+  wide?: boolean
+}
+
+interface ButtonProps extends BaseProps {
+  onClick?: (e: ClickEvent) => void
+  disabled?: boolean
+}
+
+interface AnchorProps extends BaseProps {
+  href: string
+  target?: string
+  rel?: string
+}
+
+const buttonFactory: <Props extends BaseProps>(
+  tag: Tag,
+) => React.FC<Props & InjectedProps> = tag => ({
+  size = '',
+  className = '',
+  square = false,
+  children = '',
+  prefix = '',
+  suffix = '',
+  theme,
+  ...props
+}) => {
+  const Tag = hoverable()(Base.withComponent(tag))
+
+  // prettier-ignore
+  const classNames = `${size} ${className} ${square ? 'square' : ''} ${prefix ? 'prefix' : ''} ${suffix ? 'suffix' : ''}`
+
+  return (
+    <Tag className={classNames} theme={theme} {...props}>
+      {prefix && <Prefix theme={theme}>{prefix}</Prefix>}
+      {children}
+      {suffix && <Suffix theme={theme}>{suffix}</Suffix>}
+    </Tag>
+  )
+}
+
+const ButtonComponent: React.FC<ButtonProps & InjectedProps> = buttonFactory<ButtonProps>('button')
+const ButtonAnchorComponent: React.FC<AnchorProps & InjectedProps> = buttonFactory<AnchorProps>('a')
+
+export const PrimaryButton = withTheme(ButtonComponent)
+export const PrimaryButtonAnchor = withTheme(ButtonAnchorComponent)
+
+const Base: any = styled.div`
+  ${({ theme }: InjectedProps) => {
+    const { palette, frame, size, interaction } = theme
+
+    return css`
+      display: inline-flex;
+      justify-content: center;
+      align-items: center;
+      padding: 0 ${size.pxToRem(size.space.XS)};
+      width: ${({ wide }: any) => (wide ? '100%;' : 'auto')};
+      min-width: 2rem;
+      height: 40px;
+      line-height: 40px;
+      vertical-align: middle;
+      border: none;
+      border-radius: ${frame.border.radius.m};
+      background-color: ${palette.MAIN};
+      color: #fff;
+      font-size: ${size.pxToRem(size.font.TALL)};
+      font-weight: bold;
```

Comment:
Do you need `font-weight: bold`?

Reply:
Yes, design was defined as font-weight: w6.